### PR TITLE
changes to add answers to questions provided by professor

### PR DESCRIPTION
**Pull request by Sagar-k93**
This will add all answers to answers.txt file which has git questions.

### CODE REVIEW

File: answers.txt [hunk 1]
```diff
@@ -1,4 +1,34 @@
...
+hs_err_pid*
+
+
 � What does fast-forward in Git mean? 
-� Why should you have di?erent branches? 
-� In a project how many remote Git repositories do you usually have?
\ No newline at end of file
+
+When we try to merge a commit with another, if the branch on which the merge 
+is being made hasn't diverged or changed, then git, instead of creating a new commit,
```
Comment (abhinawsarang) [inline answers.txt:22]: It does create a new commit, also merge happens but without any issue like merge conflicts.

File: answers.txt [hunk 2]
```diff
@@ -1,4 +1,34 @@
...
-� Why should you have di?erent branches? 
-� In a project how many remote Git repositories do you usually have?
\ No newline at end of file
+
+When we try to merge a commit with another, if the branch on which the merge 
+is being made hasn't diverged or changed, then git, instead of creating a new commit, 
+moves the base branch pointer to the new commit. This is called Fast-Forward in git as there is no merge and commit.
+
+
+� Why should you have different branches?
```
Comment (sakshamjhawar) [inline answers.txt:26]: You could also mention about managing a repository using multiple branches.

**Review by abhinawsarang — COMMENTED** [review]
Looks ok, please see if you can make some changes in 2nd question.

**Review by sakshamjhawar — CHANGES_REQUESTED** [review]
Please address the comments.

File: answers.txt (hunk 1 above)
Comment (Sagar-k93) [reply answers.txt:22]: Hi @abhinawsarang , please refer to the link here : [https://sandofsky.com/images/fast_forward.pdf](https://sandofsky.com/images/fast_forward.pdf). It states that a new commit is not created.

**Review by Sagar-k93 — COMMENTED** [review]
Have replied to the concerns raised, please review the pull request.

**Review by sakshamjhawar — APPROVED** [review]

**Review by abhinawsarang — APPROVED** [review]
Go ahead with the merge.

**Pull request merged**